### PR TITLE
core: frontend: animate heart icon with vehicle heartbeats, show warning icons for high temperature and disk usage

### PR DESCRIPTION
**Pull request by Williangalvani**
heartbeat

We can use different formula and behavior.
Thougths?

### CODE REVIEW

**Comment by patrickelectric** [comment]
Red looks like an error or something, maybe green ?

**Review by ES-Alexander — APPROVED** [review]
I find this hilarious, but also love that it's a thing 😍

Agreed with @patrickelectric that red looks like an error (even though it's the traditional heartbeat colour). Maybe we could do a slightly dark blue or something? Would be really cool if it could go solid red if the heartbeat is lost, or if some other health indicator fails (e.g. overheating), but understandable if that's overly complex.

**Comment by ES-Alexander** [comment]
> what about heart-broken ?

Haha, stone-cold broken heart, love it :D

**Comment by Williangalvani** [comment]
updates!
heartbea2
image

**Comment by Williangalvani** [comment]
more updates!
stuff

File: core/frontend/src/components/health/HealthTrayMenu.vue [hunk 1]
```diff
@@ -14,11 +14,37 @@
...
+          :title="`CPU temperature too high (${cpu_temperature} ºC), please cool down your vehicle!`"
+        >
+          mdi-thermometer
+        </v-icon>
+        <v-icon
+          v-if="heartbeat_age() < 3000"
           class="px-1"
-          color="white"
+          :color="`rgba(255,255,255,${0.4 + (1000-heartbeat_age())/1000}`"
+          title="MAVLink hearteats arriving as expected"
```
Comment (ES-Alexander) [inline core/frontend/src/components/health/HealthTrayMenu.vue:36]:
```suggestion
          title="MAVLink heartbeats arriving as expected"
```

File: core/frontend/src/components/health/HealthTrayMenu.vue [hunk 2]
```diff
@@ -72,9 +98,46 @@ export default Vue.extend({
...
+      const main_disk = disks?.find((sensor) => sensor.mount_point === '/')
+      return main_disk ? main_disk.available_space_B / main_disk.total_space_B / 0.01 : 0
+    },
+    disk_usage_high(): boolean {
+      return this.disk_usage_percent > 0.85
+    },
+    cpu_temperature_high(): boolean {
+      try {
+        const temp = parseFloat(this.cpu_temperature)
+        return temp > 75
```
Comment (patrickelectric) [inline core/frontend/src/components/health/HealthTrayMenu.vue:112]: We should use linux2rest flag for thermal throttle here

**Comment by Williangalvani** [comment]
> We should use linux2rest flag for thermal throttle here

That resulted in more icons for more flags...
moar

**Comment by Williangalvani** [comment]
I have mixed feelings about using the flag for temperature. while it sounds right, it potentially goes away after throttling

**Review by patrickelectric — APPROVED** [review]
image

**Pull request merged**